piano-hero/mobile/src/main/java/com/novoda/pianohero/GameModel.java

```diff
@@ -0,0 +1,105 @@
+package com.novoda.pianohero;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class GameModel implements GameMvp.Model {
+
+    private final SongSequenceFactory songSequenceFactory;
+    private final SimplePitchNotationFormatter pitchNotationFormatter;
+
+    private Sequence sequence;
+
+    GameModel(
+            SongSequenceFactory songSequenceFactory,
+            SimplePitchNotationFormatter pitchNotationFormatter) {
+        this.songSequenceFactory = songSequenceFactory;
+        this.pitchNotationFormatter = pitchNotationFormatter;
+    }
+
+    @Override
+    public RoundViewModel startGame() {
+        sequence = songSequenceFactory.maryHadALittleLamb();
+
+        checkSequenceIsSimpleElseThrow(sequence);
+
+        return createViewModel(sequence);
+    }
+
+    private void checkSequenceIsSimpleElseThrow(Sequence sequence) {
+        for (int i = 0; i < sequence.length(); i++) {
+            Notes notes = sequence.get(i);
+            if (notes.count() > 1) {
+                throw new IllegalArgumentException("Sequence contains chords, that's not simple enough");
+            }
+            for (Note note : notes.notes()) {
+                if (pitchNotationFormatter.format(note).endsWith("#")) {
```

Comment:
i think in the future Ataul will be looking at this, this PR was just a refactor to mvp